flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/operators/windowing/functions/InternalAggregateProcessAllWindowFunction.java

```diff
@@ -53,22 +57,47 @@ public InternalAggregateProcessAllWindowFunction(
 	}
 
 	@Override
-	public void apply(Byte key, final W window, Iterable<T> input, Collector<R> out) throws Exception {
+	public void open(Configuration parameters) throws Exception {
+		super.open(parameters);
 		ProcessAllWindowFunction<V, R, W> wrappedFunction = this.wrappedFunction;
-		ProcessAllWindowFunction<V, R, W>.Context context = wrappedFunction.new Context() {
-			@Override
-			public W window() {
-				return window;
-			}
-		};
+		this.ctx = new InternalProcessAllWindowContext<>(wrappedFunction);
+	}
 
+	@Override
+	public void process(Byte aByte, final W window, final InternalWindowContext context, Iterable<T> input, Collector<R> out) throws Exception {
 		final ACC acc = aggFunction.createAccumulator();
 
 		for (T val : input) {
 			aggFunction.add(val, acc);
 		}
 
-		wrappedFunction.process(context, Collections.singletonList(aggFunction.getResult(acc)), out);
+		this.ctx.window = window;
+		this.ctx.internalContext = context;
+		ProcessAllWindowFunction<V, R, W> wrappedFunction = this.wrappedFunction;
+		wrappedFunction.process(ctx, Collections.singletonList(aggFunction.getResult(acc)), out);
+	}
+
+	@Override
+	public void clear(final W window, final InternalWindowContext context) throws Exception {
+		ProcessAllWindowFunction<V, R, W> wrappedFunction = this.wrappedFunction;
+		final ProcessAllWindowFunction<V, R, W>.Context ctx = wrappedFunction.new Context() {
```

Comment:
leftover anonymous inner `Context`. This should also use `this.ctx` like in `process()`.